src/app/models/form.server.model.ts

```diff
@@ -828,6 +828,22 @@ const compileFormModel = (db: Mongoose): IFormModel => {
     ).exec()
   }
 
+  /**
+   * Retrieves all the public forms for a user which has sms verifications enabled
+   * @param userId The userId to retrieve the forms for
+   * @returns All public forms that have sms verifications enabled
+   */
+  FormSchema.statics.retrievePublicFormsWithSmsVerification = async function (
+    userId: IUserSchema['_id'],
+  ) {
+    return this.find({
+      admin: userId,
+      'form_fields.fieldType': BasicField.Mobile,
+      'form_fields.isVerifiable': true,
+      status: Status.Public,
```

Comment:
The problem with this query is that it may be expensive. Has this been benchmarked on staging?

Reply:
If this is benched to be slow, we have a minimum of 2 options:

1. Add index
2. Use current index on the `form` collection `admin_1_lastModified_-1` and programatically filter.

Reply:
resolved - worst case at present took 0.01s